lib/prelude.dx

```diff
@@ -537,6 +545,30 @@ def sq  [Mul a] (x:a) : a = x * x
 def abs [Add a, Ord a] (x:a) : a = select (x > zero) x (zero - x)
 def mod (x:Int) (y:Int) : Int = rem (y + rem x y) y
 
+instance [Floating a] Floating (n=>a)
+  exp    = map exp
+  exp2   = map exp2
+  log    = map log
+  log2   = map log2
+  log10  = map log10
+  log1p  = map log1p
+  sin    = map sin
+  cos    = map cos
+  tan    = map tan
+  sinh   = map sinh
+  cosh   = map cosh
+  tanh   = map tanh
+  floor  = map floor
+  ceil   = map ceil
+  round  = map round
+  sqrt   = map sqrt
+  pow    = \x y. for i. pow x.i y.i
+  lgamma = map lgamma
```

Comment:
I am not sure these are a great idea
One of the nice things about Dex (IMO) is not having broadcasting. (except the `broadcast` function)
Neither numpy's implict broadcasting (`sin(xs)`), nor julias explict broadcasting (`sin.(xs)`).
because `for i. sin(xs.i)` is just not that much longer and it is that much more clear (including how it generalizes)

Also we will run into problems if we generalize this to matrixes since `sin` and other expodentials are defines on matrixes.

Reply:
Hmm, would be interesting to think about this more. I don't really see the downside of using Dex's type system to do broadcasting safely. 

I was thinking about a table `a => b => c => Float` and wanting to take the `std` over the b dimension. (As now works with `add`)

`for i. for k. std for j. x.i.j.k`
versus
`std $ axis1 x`